llama_stack/providers/utils/inference/openai_compat.py

```diff
@@ -170,13 +173,37 @@ def process_completion_response(response: OpenAICompatCompletionResponse, format
 
 
 def process_chat_completion_response(
-    response: OpenAICompatCompletionResponse, formatter: ChatFormat
+    response: OpenAICompatCompletionResponse,
+    formatter: ChatFormat,
+    request: ChatCompletionRequest,
 ) -> ChatCompletionResponse:
     choice = response.choices[0]
 
     raw_message = formatter.decode_assistant_message_from_content(
         text_from_choice(choice), get_stop_reason(choice.finish_reason)
     )
+
```

Comment:
I had an issue with the `decode_assistant_message_from_content()` function above in case you have any clue: https://github.com/meta-llama/llama-stack/pull/1034#issuecomment-2649104483